internal/storagetools/check.go

```diff
@@ -0,0 +1,61 @@
+package storagetools
+
+import (
+	"bytes"
+	"fmt"
+	"math/rand"
+	"strings"
+	"time"
+
+	"github.com/wal-g/wal-g/pkg/storages/storage"
+
+	"github.com/wal-g/tracelog"
+)
+
+func HandleCheckRead(folder storage.Folder, filenames []string) {
+	objs, _, err := folder.ListFolder()
+	if err != nil {
+		tracelog.ErrorLogger.Fatalf("failed to list the storage: %v", err)
+	}
+	objNames := make(map[string]struct{}, len(objs))
+	for _, obj := range objs {
+		objNames[obj.GetName()] = struct{}{}
+	}
+	missing := make([]string, 0)
+	for _, name := range filenames {
+		if _, ok := objNames[name]; !ok {
+			missing = append(missing, name)
+		}
+	}
+	if len(missing) > 0 {
+		tracelog.ErrorLogger.Fatalf("files are missing: %s", strings.Join(missing, ", "))
+	}
+	fmt.Printf("OK")
+}
+
+func randomName(length int) string {
+	rand.Seed(time.Now().UnixNano())
+	b := make([]byte, length)
+	rand.Read(b)
+	return fmt.Sprintf("%x", b)[:length]
+}
+
+func HandleCheckWrite(folder storage.Folder) {
+	var filename string
+	for {
+		filename = randomName(32)
+		ok, err := folder.Exists(filename)
+		if err != nil {
+			tracelog.ErrorLogger.Fatalf("failed to read from the storage: %v", err)
+		}
+		if !ok {
+			break
+		}
+	}
+	err := folder.PutObject(filename, bytes.NewBufferString("test"))
+	_ = folder.DeleteObjects([]string{filename})
```

Comment:
Maybe add a warning to notify the user that we've left some garbage in the storage?

Reply:
ok